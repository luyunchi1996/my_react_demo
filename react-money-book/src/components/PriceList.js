import React from "react";
import Ionicon from "react-ionicons";
import PropTypes from "prop-types";

const PriceList = ({ items, onModifyItem, onDeleteItem }) => {
  return (
    <ul className="list-group list-group-flush">
      {items.map(item => (
        <li
          className="list-group-item d-flex 
          justify-content-between align-items-center"
          key={item.id}
        >
          <span className="col-1 ">
            <Ionicon
              className="rounded-circle"
              fontSize="24px"
              style={{ backgroundColor: "#e4e4e4", padding: "5px" }}
              color={"#000"}
              icon={item.category.iconName}
            />
          </span>
          <span className="col-5">{item.title}</span>
          <span className="col-2 font-weight-bold">
            {item.category.type === "income" ? "+" : "-"}
            {item.price}å…ƒ
          </span>
          <span className="col-2">{item.date}</span>
          <span className="col-1">
            <Ionicon
              onClick={e => {
                e.preventDefault();
                onModifyItem(item);
              }}
              className="rounded-circle"
              fontSize="24px"
              style={{ backgroundColor: "#e4e4e4", padding: "5px" }}
              color={"#000"}
              icon="ios-create-outline"
            />
          </span>
          <span className="col-1">
            <Ionicon
              onClick={e => {
                e.preventDefault();
                onDeleteItem(item);
              }}
              className="rounded-circle"
              fontSize="24px"
              style={{ backgroundColor: "#e4e4e4", padding: "5px" }}
              color={"#000"}
              icon="ios-close-circle"
            />
          </span>
        </li>
      ))}
    </ul>
  );
};
PriceList.propTypes = {
  items: PropTypes.array.isRequired,
  onModifyItem: PropTypes.func.isRequired,
  onDeleteItem: PropTypes.func.isRequired
};
PriceList.defaultProps = {
  onModifyItem: () => {},
  onDeleteItem: () => {}
};

export default PriceList;
