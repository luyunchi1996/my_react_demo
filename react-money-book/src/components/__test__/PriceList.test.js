import React from "react";
import { shallow } from "enzyme";
import PriceList from "../PriceList";
import {createSerializer} from 'enzyme-to-json';

const items = [
  {
    id: 1,
    title: "title1",
    price: 100,
    date: "2018-09-07",
    cid: "1"
  },
  {
    id: 2,
    title: "title2",
    price: 200,
    date: "2018-09-08",
    cid: "2"
  }
];
const categories = {
  "1": {
    id: 1,
    name: "旅行",
    type: "income",
    iconName: "ios-plane"
  },
  "2": {
    id: 2,
    name: "旅行",
    type: "outcome",
    iconName: "ios-plane"
  }
};

const itemsWithCategory = items.map(obj => {
  return {
    ...obj,
    category: categories[obj.cid]
  };
});

const props = {
  items: itemsWithCategory,
  onModifyItem: () => {},
  onDeleteItem: () => {}
};

let wrapper;

describe("test PriceList component ", () => {
    beforeEach(()=>{
       wrapper = shallow(<PriceList  {...props}/>)
    })

    it("should render the component to match snapshot 快照",()=>{
         expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));
         expect(wrapper).toMatchSnapshot()

    })

});
