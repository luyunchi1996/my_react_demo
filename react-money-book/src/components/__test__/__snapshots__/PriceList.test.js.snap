// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test PriceList component  should render the component to match snapshot 快照 1`] = `
<ul
  className="list-group list-group-flush"
>
  <li
    className="list-group-item d-flex 
 justify-content-between align-items-center"
    key="1"
  >
    <span
      className="col-1 "
    >
      <Ionicon
        beat={false}
        className="rounded-circle"
        color="#000"
        fontSize="24px"
        icon="ios-plane"
        rotate={false}
        shake={false}
        style={
          Object {
            "backgroundColor": "#e4e4e4",
            "padding": "5px",
          }
        }
      />
    </span>
    <span
      className="col-5"
    >
      title1
    </span>
    <span
      className="col-2 font-weight-bold"
    >
      +
      100
      元
    </span>
    <span
      className="col-2"
    >
      2018-09-07
    </span>
    <span
      className="col-1"
    >
      <Ionicon
        beat={false}
        className="rounded-circle"
        color="#000"
        fontSize="24px"
        icon="ios-create-outline"
        onClick={[Function]}
        rotate={false}
        shake={false}
        style={
          Object {
            "backgroundColor": "#e4e4e4",
            "padding": "5px",
          }
        }
      />
    </span>
    <span
      className="col-1"
    >
      <Ionicon
        beat={false}
        className="rounded-circle"
        color="#000"
        fontSize="24px"
        icon="ios-close-circle"
        onClick={[Function]}
        rotate={false}
        shake={false}
        style={
          Object {
            "backgroundColor": "#e4e4e4",
            "padding": "5px",
          }
        }
      />
    </span>
  </li>
  <li
    className="list-group-item d-flex 
 justify-content-between align-items-center"
    key="2"
  >
    <span
      className="col-1 "
    >
      <Ionicon
        beat={false}
        className="rounded-circle"
        color="#000"
        fontSize="24px"
        icon="ios-plane"
        rotate={false}
        shake={false}
        style={
          Object {
            "backgroundColor": "#e4e4e4",
            "padding": "5px",
          }
        }
      />
    </span>
    <span
      className="col-5"
    >
      title2
    </span>
    <span
      className="col-2 font-weight-bold"
    >
      -
      200
      元
    </span>
    <span
      className="col-2"
    >
      2018-09-08
    </span>
    <span
      className="col-1"
    >
      <Ionicon
        beat={false}
        className="rounded-circle"
        color="#000"
        fontSize="24px"
        icon="ios-create-outline"
        onClick={[Function]}
        rotate={false}
        shake={false}
        style={
          Object {
            "backgroundColor": "#e4e4e4",
            "padding": "5px",
          }
        }
      />
    </span>
    <span
      className="col-1"
    >
      <Ionicon
        beat={false}
        className="rounded-circle"
        color="#000"
        fontSize="24px"
        icon="ios-close-circle"
        onClick={[Function]}
        rotate={false}
        shake={false}
        style={
          Object {
            "backgroundColor": "#e4e4e4",
            "padding": "5px",
          }
        }
      />
    </span>
  </li>
</ul>
`;
